src/staking/block_validator.cpp

```diff
@@ -178,9 +179,9 @@ class BlockValidatorImpl : public AbstractBlockValidator {
       uint256 *snapshot_hash_out       //!< [out] The snapshot hash extracted from the scriptSig
       ) const override {
 
-    // check that there are transactions
-    if (block.vtx.empty()) {
-      return BlockValidationResult(Error::NO_TRANSACTIONS);
+    // check block size limits
+    if (!CheckBlockWeight(block)) {
```

Comment:
It is not clear that `CheckBlockWeight` checks `vtx` size. Why not keep the check resulting `Error::NO_TRANSACTIONS` and drop the check from `CheckBlockWeight`?
And then `INVALID_BLOCK_LENGTH` could have a more sane name, e.g. `INVALID_BLOCK_SIZE`.

Reply:
Renamed and commented the function in https://github.com/dtr-org/unit-e/pull/1106/commits/8a8513351930df4df6c64ba819b4cb344a05f46c